SAMLPortal/Middlewares/SetupAssistantMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using SAMLPortal.Models;
+
+namespace SAMLPortal.Middlewares
+{
+	public class SetupAssistantMiddleware
+	{
+		private readonly RequestDelegate _next;
+
+		public SetupAssistantMiddleware(RequestDelegate next)
+		{
+			_next = next;
+		}
+
+		public Task Invoke(HttpContext httpContext)
+		{
+			var configurationStep = GlobalSettings.GetInt("CONFIG_SETUPASSISTANT_STEP");
+
+			if (!httpContext.Request.Path.ToString().Contains("Setup") && configurationStep < 6)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'httpContext' before using it.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)